### ISSUE TITLE
Wallet transaction history incorrect when multiple wallets IDs with different SKY in single transaction

### ISSUE THREAD

**Issue by BigOokie** [skycoin/skycoin#1748]
Appears to be an issue with the transaction history and transaction detail UIs in the new SkyCoin Wallet v0.24.0.  I'm running the version for MacOS downloaded from the SkyCoin website.

The scenario appears to arise (or at least is visible) when a single received transaction contains multiple wallet IDs with different values for each - one of which is your wallet.  

As an example, request to `http://127.0.0.1:51080/api/v1/explorer/address?address={walletID}` returns a JSON response - where the outputs are listed as an array indexed from 0 (zero) I believe. In the Wallet UI, the transactions are listed starting with an index of 1.

Looking at the raw data in the JSON array, I can see X-SKY was received for my wallet ID. In the Wallet UI however, it shows Y-SKY received.  In the raw JSON data, I can see this value (Y-SKY) is the value associated with the wallet ID before and after my wallet ID.

I note that the online explorer shows the correct value.

**Comment by BigOokie** [comment]
Note this does not impact the balance of the Wallet - just the transaction history from my (limited) testing.

**Comment by Senyoret1** [comment]
I'm sorry, but I tested it in several ways and I could not reproduce the error.

One of my tests was to send, in one transaction from a single address, different amounts to several addresses (just one of them mine, and it was neither the first, nor the second nor the last). When checking the history in the wallet, the the history showed the correct amount of coins received.

Probably the fact that `/api/v1/explorer/address` returns the transactions in reverse order could have caused some confusion, or maybe I did not fully understand the problem.
